Mage.Sets/src/mage/cards/b/BrineHag.java

```diff
@@ -77,22 +77,10 @@ public boolean apply(Game game, Ability source) {
             return false;
         }
 
-        List<Permanent> list = new ArrayList<>();
-        for (UUID playerId : game.getState().getPlayersInRange(controller.getId(), game)) {
-            Player player = game.getPlayer(playerId);
-            if (player == null) {
-                continue;
-            }
-
-            for (Permanent creature : game.getBattlefield().getAllActivePermanents(StaticFilters.FILTER_PERMANENT_CREATURE, playerId, game)) {
-                if (sourcePermanent.getDealtDamageByThisTurn().contains(new MageObjectReference(creature.getId(), game))) {
-                    list.add(creature);
-                }
-            }
-        }
-        if (!list.isEmpty()) {
+        Set<MageObjectReference> set = new HashSet<>(sourcePermanent.getDealtDamageByThisTurn());
+        if (!set.isEmpty()) {
             FilterCreaturePermanent filter = new FilterCreaturePermanent();
-            filter.add(new PermanentInListPredicate(list));
+            filter.add(new PermanentReferenceInCollectionPredicate(set));
```

Comment:
You were also replaced a usage of `PermanentInListPredicate` by `PermanentReferenceInCollectionPredicate` in some cards, but miss it in another cards without any mention in the PR description (looks like not finished refactor/replace for the new class).

![shot_230718_092021](https://github.com/magefree/mage/assets/8344157/f79fe364-089c-412c-a935-6039a438a814)

Reply:
It only needed to be replaced for effects that aren't instantaneous/one-shot. I could replace all of them I suppose, but the others are fully working with the previous class.